### ISSUE TITLE
Error on writing a XML/Interlis file when parameter "Validate" is "Yes" 

### ISSUE THREAD

**Issue by pasimo** [claeis/ili2fme#18]
The following error occurred during the generation of an XML-Interlis catalogue, although finally a proper and valid file has been created.

![Error_ili2fme_when_validate_is_on](https://user-images.githubusercontent.com/37699598/114719429-5cc35780-9d37-11eb-852b-da3142a71a1b.png)

Best regards
Pascal

**Comment by beistehen** [comment]
How do you verify that the file is 

> proper and valid

?

Can you provide the data for testing?

**Comment by pasimo** [comment]
I checked with ilivalidator 1.11.9

Attachments: model and interlis file

Von: Stefan Henrich ***@***.***>
Gesendet: Mittwoch, 14. April 2021 15:57
An: claeis/ili2fme ***@***.***>
Cc: Imoberdorf Pascal BAZL ***@***.***>; Author ***@***.***>
Betreff: Re: [claeis/ili2fme] Error on writing a XML/Interlis file when parameter "Validate" is "Yes" (#18)


How do you verify that the file is

proper and valid

?

Can you provide the data for testing?

**Comment by beistehen** [comment]
(I guess you have to upload attachments via Github website...)

**Comment by pasimo** [comment]
[ObstacleCollection_Catalogues_ObstacleType.zip](https://github.com/claeis/ili2fme/files/6311606/ObstacleCollection_Catalogues_ObstacleType.zip)

**Comment by pasimo** [comment]
Ok, done...

**Comment by beistehen** [comment]
Could it be that you wrote multiple catalogs (e. g. multiple xml files) at once in FME? This would explain the first error in the log file

> BID b2 ... already exists...

and therefore the subsequent errors in the log.

**Comment by pasimo** [comment]
Therefore I find the error message is a little weird. Only a single basket exists and only one XML catalogue is written.

[ili2fme_Validate-Parameter.zip](https://github.com/claeis/ili2fme/files/6315814/ili2fme_Validate-Parameter.zip)

**Comment by beistehen** [comment]
I agree, this is weird. Even the following simple workspace raises the BID error (for an even simpler model):
![grafik](https://user-images.githubusercontent.com/26199857/114859774-7cb45300-9deb-11eb-9f69-ada7fe56415f.png)

![grafik](https://user-images.githubusercontent.com/26199857/114860627-85595900-9dec-11eb-9364-fd30c22616cc.png)

It looks like validation is invoked twice.
And it also looks like the built-in validation option is rarely used...
@claeis any ideas?

**Comment by KTAI-GIS** [comment]
As mentioned by Stefan the problem can be reproduced by his simple XTF_BASKET example.
Reproduced on FME2020 (x86) and FME2021 (x86) with JAVA RTE 1.8.0_301.
The problem does not appear on FME2019 (x86)
